app/src/main/java/com/android/sample/ui/swipePage/SwipePage.kt

```diff
@@ -89,169 +106,193 @@ fun SwipePage(navigationActions: NavigationActions) {
-private fun setNextImage(current: Int): Int {
-  if (current == R.drawable.burger) return R.drawable.salad
-  return R.drawable.burger
-}
-
 /**
- * Description of the recipe
+ * Composable for the Image Description
  *
- * @param isDescriptionVisible - Tracks if the description should be visible
- * @param modifier - Modifier for the column
+ * @param name - Recipe Name
+ * @param tag - Recipe Tag
+ * @param modifier - Modifier
  */
 @Composable
-private fun Description(isDescriptionVisible: Boolean, modifier: Modifier) {
+private fun ImageDescription(name: String, tag: String, modifier: Modifier) {
 
-  Column(modifier = Modifier.testTag("recipeDescription").padding(16.dp)) {
+  Column(modifier = Modifier.padding(16.dp)) {
+    Row(
+        modifier =
+            modifier.fillMaxWidth().testTag("draggableItem"), // Ensure the Row takes up full width
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween) {
 
-    // Display Recipe Name
-    Text(
-        modifier = modifier,
-        text = "Pasta",
-        style = MaterialTheme.typography.bodyLarge,
-        color = MaterialTheme.colorScheme.onSecondary,
-    )
-    if (!isDescriptionVisible) {
-      // Display "..." if the description is not visible
-      Text(
-          modifier = modifier.testTag("displayDescription"),
-          text = "...",
-          style = MaterialTheme.typography.bodyLarge,
-          color = MaterialTheme.colorScheme.onSecondary,
-      )
-    }
-    Column(modifier = Modifier.verticalScroll(rememberScrollState())) {
-      // Display with animation Recipe Description
-      AnimatedVisibility(
-          visible = isDescriptionVisible, enter = fadeIn(animationSpec = tween(500))) {
-            Text(
-                modifier = Modifier.testTag("RecipeEntireDescription"),
-                text = getRecipeDescription(),
-                style = MaterialTheme.typography.bodyMedium,
-                color = MaterialTheme.colorScheme.onSecondary,
-            )
-          }
-
-      Spacer(modifier = Modifier.padding(5.dp))
+          // Display Recipe Name
+          RecipeDescription(name, "4.5")
+        }
+    Spacer(modifier = Modifier.padding(2.dp))
+    Row(verticalAlignment = Alignment.CenterVertically) { Tag(tag) }
+  }
+}
 
-      // Display Recipe Timing
-      Row(
-          modifier = Modifier,
-          verticalAlignment =
-              Alignment.CenterVertically // Aligns the icon and text vertically centered
-          ) {
-            Icon(
-                painter = painterResource(R.drawable.timer),
-                contentDescription = "recipes timing",
-                modifier = Modifier.size(24.dp) // Adjust size to match the text size
-                )
+/**
+ * Composable for the Recipe Description
+ *
+ * @param name - Recipe Name
+ * @param rate - Recipe Rate
+ */
+@Composable
+fun RecipeDescription(name: String, rate: String) {
```

Comment:
This is a well-structured composable! A few minor suggestions:

1. **Magic Number for `Spacer` Padding**: Consider replacing `5.dp` with a named constant for better readability and flexibility.

2. **Modifier Chaining**: Instead of applying `Modifier.size(24.dp)` and then adding `.testTag()`, it might be cleaner to chain them using `.then(Modifier)` for clarity.

3. **Font Size**: The `fontSize = 24.sp` is hardcoded. It might be better to use `MaterialTheme`’s typography to ensure consistency across the app.

Otherwise, this looks great and does a good job of keeping the layout clean and aligned.